shell/platform/embedder/embedder.h

```diff
@@ -434,6 +435,52 @@ typedef struct {
   BoolPresentInfoCallback present_with_info;
 } FlutterOpenGLRendererConfig;
 
+/// Handle for the MTLDevice.
+typedef const void* FlutterMetalDevice;
+
+/// Handle for the MTLCommandQueue.
+typedef const void* FlutterMetalCommandQueue;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterMetalTexture).
+  size_t struct_size;
+  /// Embedder provided unique identifier to the texture buffer. Given that the
+  /// `texture` handle is passed to the engine to render to, the texture buffer
+  /// is itseld owned by the embedder. This `texture_id` is then also given to
+  /// the embedder in the present callback.
+  int64_t texture_id;
+  /// Handle to the MTLTexture that is owned by the embedder. Engine will render
+  /// the frame into this texture.
+  void* texture;
+} FlutterMetalTexture;
+
+/// Callback for when a metal texture is requested.
+typedef void (*FlutterMetalTextureCallback)(
+    void* /* user data */,
+    const FlutterFrameInfo* /* frame info */,
+    FlutterMetalTexture* /* texture out */);
```

Comment:
If the embedder is giving the engine a texture, how can it use a texture as an out parameter? Should this be

```
typedef FlutterMetalTexture (*FlutterMetalTextureCallback)(
    void* /* user data */,
    const FlutterFrameInfo* /* frame info */);
```

And, later, in the present callback, the same `FlutterMetalTexture` be given back to the embedder for presentation.

Reply:
In my current implementation the embedder fills out the texture_out arg. But you're right it could just be returning FlutterMetalTexture instead.